doc/list.md

```diff
@@ -1 +1,195 @@
 # List
+
+A `List` looks similar to an [Array](Array.md) but it is a pure Lean inductive type that does not
+have an optimized runtime other than what the Lean compiler builds from the List implementation.
+
+```lean
+# namespace hidden
+inductive List (α : Type u) where
+  | nil : List α
+  | cons (head : α) (tail : List α) : List α
+# end hidden
+```
+
+A list then is defined recursively through concatenation of some `head` item with an existing `tail`
+via the `cons` constructor.
+
+## Literals
+
+You can create lists in several ways. You can start with a small list by listing consecutive values between
+`[` and `]` and separated by commas, as shown in the following examples.
+
+```lean
+#check [1, 2, 3] -- List Nat
+
+#check #[1, 2, 3] -- Array Nat - arrays have a # prefix.
+
+#check [] -- List ?m  -- unresolved type.
+#check List.nil (α := String) -- List String - specify the type parameter
+
+#eval List.cons "a" ["b", "c", "d"]     -- ["a", "b", "c", "d"]
+```
+
+Notice that Lean is able to infer the list element type in these cases, createing `List nat` and `List String`.
+
+Notice also that `[]` is short hand for `List.nil` and that it is missing any list element type. To
+resolve this you can write the following:
+
+There is special syntax for `cons`, namely, `::` as follows:
+
+```lean
+#eval 0 :: [1, 2, 3]        -- [0, 1, 2, 3]
+```
+
+The type of the list elements is inferred from the literals used and must be consistent.
```

Comment:
There is nothing special about `List.cons`. We should explain the type system in a different part of the manual once and for all.

Reply:
I don't understand this comment.  `List.cons` is clearly specific to lists, what is general type system info here?

Reply:
I was referring to the sentence 

> The type of the list elements is inferred from the literals used and must be consistent.

If one understands how the type system works, the sentence is redundant.
The second problem is that it uses non-standard terminology, we don't say the types are consistent.
I think all discussions regarding typing should go into a different session.